src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -209,38 +190,35 @@ private function createRelationsForItem($item, $formattedData)
      * Those will be nested accordingly in this relation array, so address relation will have a nested relation with country.
      *
      *
-     * @param  array  $input  The form data.
-     * @return array The formatted relation data.
+     * @param  array  $input  The form input.
+     * @return array The formatted relation details.
      */
-    private function getRelationDataFromFormData($data)
+    private function getRelationDetailsFromInput($input)
```

Comment:
Much better name! 🙏

Reply:
@pxpm it looks like this _always_ returns an array with one item ("relations"), that is an array that holds the relations. Am I mistaken? Any reason to do this?

```php
// instead of returning
[
    'relations' => [
         'comments' => [],
         'stars' => [],
    ]
]
// wouldn't it be simpler to return this?
[
     'comments' => [],
     'stars' => [],
]
```

If that's possible, I think this would simplify the `Arr:get()` and `Arr:set()` too, no more need for the `.relations.` prefix. That would make the code more readable.